### PR TITLE
Improve createdEvent subscriber to allow overiddes in archetypes.multilingual

### PR DESCRIPTION
**Pull request by mpeeters**
We currently encounter 2 problems when we try to translate archetypes content types on Plone 4.3.
 * An error with Archetypes temporary objects (portal_factory)
 * A problem with the translation group on translated objects

This PR fix the error with Archetypes temporary objects and add the possibility to overrides the createdEvent subscriber in archetypes.multilingual to fix the issue with the translation group.

We have an other PR for archetypes.multilingual

cc @bsuttor

### CODE REVIEW

**Comment by datakurre** [comment]
@mpeeters @bsuttor Now when you are familiar with archetypes.multilingual, would you be able to look into howto make https://github.com/plone/plone.app.multilingual/pull/154 compatible with this and archetypes.multilingual.

See also #120 for discussion.

TL;DR; We want to get rid of session and use custom ++translation++SOURCE-UUID -traverse instead. That traverse annotates request (not session) with required translation info (e.g. tg and source lang). It works for DX, because ++add++-forms will post to that traverser address by default, but Archetypes may need some other magic, because of portal_factory.

**Comment by bsuttor** [comment]
Ok, I can check that, 
But at this moment, I can't merge branch issue-120-remove-session because we change way of subscriber.py works. We refactor the createdEvent subscriber to allow overrides in archetypes.multilingual (you can see that here : https://github.com/IMIO/plone.app.multilingual/commit/0c6dc36906e3da1f729ec05d1eaeb6b42b496a5f)

I don't know how progress. Can you merge our PR into issue-120-remove-session or can I do it ? Or merging our PR on master and we merge conflict after ?

**Comment by datakurre** [comment]
@bsuttor I cannot merge this into #154, because I'm not yet familiar with at.multilingual. I think, you would be the best one to re-refactor subscriber.py in issue-120-remove-session to support at.multilingual. Maybe by creating a new branch from issue-120-remove-session and trying to rebase this on top of that.

AFAIK, the major changes in issue-120-remove-session are:

- removal of "has_pam_old_lang_in_form" condition, because that approach was abandoned
- removal of session (and using request.translation_info instead).

Yet, I still expect changes in ++addtranslation++-traverser to cause issues with AT, as described earlier.

Again, I'm sorry about #154 conflicting with this, but I think we all agree that PAM should be stateless and not require shared session data between instances.

**Comment by mpeeters** [comment]
see #160
